components/public-api-server/cmd/root.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/gitpod-io/gitpod/public-api/config"
+)
+
+var (
+	// ServiceName is the name we use for tracing/logging
+	ServiceName = "public-api-server"
+	// Version of this service - set during build
+	Version = ""
+)
+
+var rootOpts struct {
+	CfgFile string
+	JsonLog bool
+	Verbose bool
+}
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   ServiceName,
+	Short: "serves public API services",
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		log.Init(ServiceName, Version, rootOpts.JsonLog, rootOpts.Verbose)
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
+	rootCmd.PersistentFlags().StringVar(&rootOpts.CfgFile, "config", "", "config file")
```

Comment:
IMO, this shouldn't be a persistent flag on the root command as it doesn't really have any meaning for the root command. It's only relevant for the `run` sub-command. Due to this, it also prints somewhat misleading command help text:
```
go run main.go    
serves public API services

Usage:
  public-api-server [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  run         starts the service

Flags:
      --config string   config file
  -h, --help            help for public-api-server
  -j, --json-log        produce JSON log output on verbose level (default true)
  -v, --verbose         Enable verbose JSON logging

Use "public-api-server [command] --help" for more information about a command.
```